rust/main/hyperlane-base/src/contract_sync/cursors/metrics.rs

```diff
@@ -0,0 +1,65 @@
+use prometheus::IntGaugeVec;
+
+use crate::CoreMetrics;
+
+/// Struct encapsulating prometheus metrics used by SequenceAware and RateLimited cursors.
+#[derive(Debug, Clone)]
+pub struct CursorMetrics {
+    /// Current block of the cursor.
+    /// Used by both sequence aware and rate limited cursors.
+    /// Labels:
+    /// - `event_type`: the event type the cursor is indexing. Could be anything implementing `Indexable`.
+    /// - `chain`: Chain the cursor is collecting data from.
+    /// - `cursor_type`: The type of cursor. E.g. `forward_sequenced`, `backward_sequenced`, `forward_rate_limited` or `backward_rate_limited`.
+    pub cursor_current_block: IntGaugeVec,
+
+    /// Current sequence of the cursor.
+    /// Only used by sequence aware cursors.
+    /// Labels:
+    /// - `event_type`: the event type the cursor is indexing. Could be anything implementing `Indexable`.
+    /// - `chain`: Chain the cursor is collecting data from.
+    /// - `cursor_type`: The type of cursor. E.g. `forward_sequenced`, `backward_sequenced`, `forward_rate_limited` or `backward_rate_limited`.
```

Comment:
I wonder if values `forward_rate_limited` and `backward_rate_limited` are valid for this metric?

Reply:
`forward_rate_limited` is indeed valid as it's the only mode for the RateLimitedCursor, as seen [here](https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/4892#issuecomment-2504574528)

`backward_rate_limited` removed in both occurrences.